packages/hardhat-core/src/internal/core/providers/accounts.ts

```diff
@@ -53,22 +65,24 @@ export class LocalAccountsProvider extends ProviderWrapperWithChainId {
     const params = this._getParams(args);
 
     if (args.method === "eth_sign") {
-      const [address, data] = params;
-
-      if (address !== undefined) {
-        if (data === undefined) {
-          throw new HardhatError(ERRORS.NETWORK.ETHSIGN_MISSING_DATA_PARAM);
+      if (params.length > 0) {
+        const [address, data] = validateParams(params, rpcAddress, rpcData);
```

Comment:
We validate things using the types here now. The main goal was to make sure that we keep this and Hardhat Network in sync, specially now that there are multiple transaction types.

This is where the API is made sightly stricter. But as I mentioned, Hardhat and Hardhat Network were always meant to be strict.